Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,31 @@
+# create a lock using `flock (2)`. A name is required as first argument.
+# the lock will be automatically unlocked when the shell process quits.
+lock() {
+  local name="$1"
+  local lock_file="/tmp/homebrew-${name}.lock"
```

Comment:
@UniqMartin There are two solutions, one is to put LOCK_DIR into HOMEBREW_LIBRARY. The second is to do some thing like
```bash
lock_file="/tmp/homebrew${HOMEBREW_REPOSITORY//\//-}-${name}.lock"
```

Which one do you think is better?